python/pyspark/pandas/series.py

```diff
@@ -4307,17 +4307,20 @@ def keys(self) -> "ps.Index":
         """
         return self.index
 
-    # TODO: 'regex', 'method' parameter
+    # TODO: introduce 'method', 'limit', 'in_place'; fully support 'regex'
     def replace(
         self,
         to_replace: Optional[Union[Any, List, Tuple, Dict]] = None,
         value: Optional[Union[List, Tuple]] = None,
-        regex: bool = False,
+        regex: Union[str, bool] = False,
     ) -> "Series":
         """
         Replace values given in to_replace with value.
         Values of the Series are replaced with other values dynamically.
 
+        .. note:: For partial pattern matching, the replacement is against the whole string,
+            which is different from pandas'.
```

Comment:
Can we a bit more elaborate why our behavior is different from pandas'?

Reply:
Certainly. Added.